### DISCUSSION TITLE
General-Questions

### DISCUSSION THREAD

**Discussion by eskeyaar** [microsoft/InnerEye-DeepLearning#485] [Q&A] [answered]
Hi,
I have few questions which i think i may not understood properly, please correct me if am wrong, wanted to explore more this toolkit.

1. So This toolkit can be run completely in local machine with some GPU power to segment the model? 
2. this toolkit already has the AI power to auto-segment most of main organs/tumors? if i want to run these on my local machine only?
3. What does both InnereyeGateway or inference mean? is it like external way of linking where the model segmentation are validated/ from Azure site?
4. Does this toolkit when complied offer GUI or it will be Command line interface? or both does it support?
5. Does this AI segment all organs or limited to any?
Thanks in Advance.
 

[AB#4102](https://innereye.visualstudio.com/60ce1777-00d6-4015-82bc-488a0c00202f/_workitems/edit/4102)

**Comment by eskeyaar** [comment]
anyone?

**Comment by ant0nsc** [comment, marked as answer]
Hey @eskeyaar - seems my answer did not get transferred over when I migrated that from an issue to a discussion.

1) yes, you can run that on any GPU machine to build a model and later segment data
2) No, there are no pre-built models. You would have to provide suitable data for your specific use-case. "Suitable data" means that you have for example CT or MR scans, and then you draw each organ using an annotation tool. From a large set of those, you can then build a model using this toolkit.
3) Please check the details [here](https://github.com/microsoft/InnerEye-DeepLearning/blob/main/docs/deploy_on_aml.md)
4) There is no GUI
5) As mentioned above, there are no pre-built models. And it is quite likely very difficult to build one that would really segment all organs.

The most important point here is really that this toolkit is NOT a general purpose pre-built AI to segment body parts.

I'd suggest reading some tutorial on the subject of medical image segmentation. A bit dated, but still valuable is [this one](http://people.csail.mit.edu/rameshvs/MEC2014/)

**Comment by eskeyaar** [comment]
@ant0nsc  - Thanks for the Answers. 
Since you have mentioned that there are no pre build models, need to provide suitable data, which means i have to draw/classify each organ/tumor/bone first and then it can able to identify? im not understanding it properly. 

Also, This AI is able to detect only models like Lungs,prostate, head&neck , brain only?  
if you can explain that would very helpful for me. 

Thanks your support.

**Reply by ant0nsc** [reply]
I've updated my answer, please reply to the answer above if you have any further questions. Please tick "answered" if your questions are covered. I'd suggest to read a bit of background material on medical image segmentation.

**Reply by eskeyaar** [reply]
Brilliant. Thanks Anton, Will go through those tutorial. Thanks again!
